### ISSUE TITLE
 Cannot read property 'depricated' of undefined

### ISSUE THREAD

**Issue by TAGC** [mrin9/RapiPdf#34]
I'm trying to generate a PDF from a Swagger document containing a deprecated endpoint and the operation fails. I see this in the console output:

```
rapipdf-min.js:45609 Uncaught (in promise) TypeError: Cannot read property 'depricated' of undefined
    at rapipdf-min.js:45609
    at Array.map (<anonymous>)
    at M (rapipdf-min.js:45609)
    at B (rapipdf-min.js:45609)
    at rapipdf-min.js:45609
    at Array.map (<anonymous>)
    at D (rapipdf-min.js:45609)
    at A (rapipdf-min.js:45609)
    at A.next (<anonymous>)
    at y (rapipdf-min.js:45609)
```

I assume that the error is caused by this line in the RapiPdf source code:

https://github.com/mrin9/RapiPdf/blob/4bf0538a0d6bda78a3ae551e78228c3e452dc4e6/src/pdf-parts-gen.js#L137

Should it be `paramSchema.deprecated` instead?

**Comment by TAGC** [comment]
Nevermind, I see the problem:

![](https://i.imgur.com/gf2zFtS.png)

The document I'm using has the following content:

```json
  "paths": {
    "/api/Quote": {
      "post": {
        "tags": [
          "Quote"
        ],
        "summary": "Generates a new quote in the Sales Office system.",
        "parameters": [
          {
            "name": "application",
            "in": "header"
          },
          {
            "name": "user-key",
            "in": "header"
          }
        ],
```

RapiPdf seems to require that these have a `schema ` property, and according to the [OpenApi 3 spec](https://swagger.io/docs/specification/describing-parameters/#header-parameters) it seems that all parameters are required to have a `schema` or `content` property in which case that seems like an error within the document and not RapiPdf.

I manually create these headers in my backend code so I should be able to fix this, test it works and then close this ticket:

```cs
private static readonly OpenApiParameter ApplicationParameter = new OpenApiParameter
{
    Name = "application",
    In = ParameterLocation.Header,
    Required = false
};

private static readonly OpenApiParameter UserKeyParameter = new OpenApiParameter
{
    Name = "user-key",
    In = ParameterLocation.Header,
    Required = false
};
```

**Comment by TAGC** [comment]
The PDF was generated once I fixed up the document that got created:

```cs
private static readonly OpenApiParameter ApplicationParameter = new OpenApiParameter
{
    Name = "application",
    In = ParameterLocation.Header,
    Required = false,
    Schema = new OpenApiSchema()
};

private static readonly OpenApiParameter UserKeyParameter = new OpenApiParameter
{
    Name = "user-key",
    In = ParameterLocation.Header,
    Required = false,
    Schema = new OpenApiSchema()
};
```

Is `paramSchema.depricated` still meant to be `paramSchema.deprecated` though?

**Comment by mrin9** [comment]
yes `paramSchema.depricated` should be `paramSchema.deprecated 
thanks for pointing it out, will get it fixed in our next patch

**Comment by mrin9** [comment]
fixed in release `2.1.1`
